src/packet/packet_data.rs

```diff
@@ -0,0 +1,43 @@
+use super::header::HeaderParser;
+use super::RawPacketData;
+
+use std::io::{Result};
+
+#[derive(Debug)]
+/// Contains the raw data this packet exists of. Note that an packet can be divided into seperate fragments
+pub struct PacketData
+{
+    parts: Vec<RawPacketData>
+}
+
+impl PacketData {
+    pub fn new() -> PacketData
+    {
+        PacketData { parts: Vec::new() }
```

Comment:
Since, we have a fixed size packet data do you think we could make this with_capacity?

Reply:
Yea think we can do that